distributed/deploy/adaptive.py

```diff
@@ -188,14 +188,19 @@ def _adapt(self):
 
         self._adapting = True
         try:
-            if self.should_scale_up():
-                kwargs = self.get_scale_up_kwargs()
-                f = self.cluster.scale_up(**kwargs)
-                if gen.is_future(f):
-                    yield f
-
-            if self.should_scale_down():
-                yield self._retire_workers()
+            should_scale_up = self.should_scale_up()
+            should_scale_down = self.should_scale_down()
+            if should_scale_up and should_scale_down:
+                logger.info("Attempting to scale up and scale down simultaneously.")
```

Comment:
Given that we know that this can occur in normal operation we should probably demote this to `logger.debug` level.  Otherwise users might think that there is something wrong with their system, rather than with our algorithms :)

Reply:
Maybe I'm overthinking it, but we should change the algorithm so that it *shouldn't* happen in normal operation. Seeing this message highlights that there is a flaw in the scaling algorithm being used, whether that be a subclass of Adaptive or the default version.

With regards to fixing the algorithm in Adaptive, so that we don't see it in normal use, this should mostly be fixed by #1609, although I think we may need a couple more changes to finish eliminate the corner cases: (these should probably go in a separate PR)

1. Don't scale up to have more cores than  there are tasks.
2. If we see should_scale_up and should_scale_down both True, try forcing a steal rebalance, and check again.

Reply:
> With regards to fixing the algorithm in Adaptive, so that we don't see it in normal use, this should mostly be fixed by #1609, although I think we may need a couple more changes to finish eliminate the corner cases:

OK, if this is true then I agree that leaving this at INFO probably makes sense